packages/core/utils/getRelativeUrl.js

```diff
@@ -20,15 +20,23 @@ export default function getRelativeUrl(url) {
       pathname: urlPath,
     } = urlObj;
     const {
+      protocol,
       host,
       query,
       hash,
     } = urlObj;
 
+    // Add trailing slashes and remove host only under below conditions.
     if (
-      host === window.location.host ||
-      host.includes(window.location.host) ||
-      window.location.host.includes(host)
+      (
+        host === window.location.host ||
+        host.includes(window.location.host) ||
+        window.location.host.includes(host)
+      ) &&
+      (
+        ('http:' === protocol || 'https:' === protocol) &&
+        'rss' !== url.split('.').pop()
```

Comment:
Instead of checking for `.rss` extensions, could we run it through the regex lists defined for the proxy? In theory that would capture things more accurately.

Additionally, maybe we just check if there's a file extension at all?